CONTRIBUTING.rst

```diff
@@ -0,0 +1,480 @@
+issue you're finding, or that have similar ideas for changes or additions.
+Send a message to the mailing list to introduce and discuss your idea with the
+development community.
+
+It's always a good practice to search for existing or related issues before
+submitting your own. When you submit an issue (bug or feature request), the
+triage team will review and comment on the submission, typically within a few
+business days.
+
+You can find all `open pull requests`_ on GitHub and open `Zephyr Project
+Issues`_ in Jira.
+
+Development Tools and Git Setup
+*******************************
+
+Signed-off-by
+=============
+
+The name in the commit message ``Signed-off-by:`` line and your email must
+match the change authorship information. Make sure your *.git/config* is set
+up correctly::
+
+   $ git config --global user.name "David Developer"
+   $ git config --global user.email "david.developer@company.com"
+
+gitlint
+=========
+
+When you submit a pull request to the project, a series of checks are
+performed to verify your commit messages meet the requirements, the same step
+done during the CI process can be performed locally using the the *gitlint*
+command.
+
+Install gitlint and run it locally in your tree and branch where your patches
+have been committed::
+
+     $ sudo pip3 install gitlint
+     $ gitlint
+
+Note, gitlint only checks HEAD (the most recent commit), so you should run it
+after each commit, or use the ``--commits`` option to spefify a commit range
+covering all the development patches to be submitted.
+
+sanitycheck
+===========
+
+To verify that your changes did not break any tests or samples, please run the
+``sanitycheck`` script locally before submitting your pull request to GitHub. To
+run the same tests the CI system runs, follow these steps from within your
+local Zephyr source working directory::
+
+    $ source zephyr-env.sh
+    $ make host-tools
+    $ export PREBUILT_HOST_TOOLS=${ZEPHYR_BASE}/bin
+    $ export USE_CCACHE=1
+    $ ./scripts/sanitycheck
+
+The above will execute the basic sanitycheck script, which will run various
+kernel tests using the QEMU emulator.  It will also do some build tests on
+various samples with advanced features that can't run in QEMU.
+
+We hightly recommend you run these tests locally to avoid any CI failures.
+Using CCACHE and pre-built host tools is optional, however it speeds up the
+execution time considerably.
+
+
+Coding Style
+************
+
+Use these coding guidelines to ensure that your development complies with the
+project's style and naming conventions.
+
+.. _Linux kernel coding style:
+   https://kernel.org/doc/html/latest/process/coding-style.html
+
+
+In general, follow the `Linux kernel coding style`_, with the
+following exceptions:
+
+* Add braces to every ``if`` and ``else`` body, even for single-line code
+  blocks. Use the ``--ignore BRACES`` flag to make *checkpatch* stop
+  complaining.
+* Use hard tab stops. Set the tab width to 8 spaces.
```

Comment:
Actually, come to think of it, hard tab stops with 8 space tabs is not an exception to the Linux style; it's the same on Linux:

https://www.kernel.org/doc/html/v4.12/process/coding-style.html#indentation

This line should be removed.